File: BettingEngineServer/BettingEngineServer/Classes/Market.cs
using System.Collections.Generic;
using BettingEngineServer.Interfaces;

namespace BettingEngineServer.Classes
{
    public class Market : IEntity
    {
        public Market()
        {
            MarketBets = new List<Bet>();
        }
        public string Id { get; set; }
        public decimal MarketProbability { get; set; }
        public string MarketDescription { get; set; }
        public string EventId { get; set; }
        public List<Bet> MarketBets { get; set; }     

        public decimal MarketOdds
        {
            get
            {
                if (MarketProbability == 0) return 0;
                return 1 / MarketProbability;
            }
        }

        
        
    }
}

File: betting-engine-client/src/app/classes/market.ts
import {Bet} from './bet';

export class Market {
  constructor() {
    this.marketBets = [];
  }

  id: string;
  marketProbability: number;
  marketDescription: string;
  eventId: string;
  marketBets: Bet[];

  public getMarketOdds() {
    if (this.marketProbability === 0) {
      return 0;
    }
    return 1 / this.marketProbability;
  }
}
